eZ/Publish/Core/Persistence/TransformationProcessor/PcreCompiler.php

```diff
@@ -273,4 +273,18 @@ protected function compileCharacter($char)
                 throw new RuntimeException("Invalid character definition: $char");
         }
     }
+
+    /**
+     * Converts a hexadecimal string to a decimal number.
+     *
+     * In compare to standard hexdec function it will ignore any non-hexadecimal characters
```

Comment:
```suggestion
     * In comparison to standard hexdec function it will ignore any non-hexadecimal characters
```
Side note: IMHO in the next major (for eZ Platform 4.0 ATM) we should throw InvalidArgumentException if hex value contains some garbage.